packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelProvider.cs

```diff
@@ -517,12 +515,12 @@ p.Property is null
                     }
                     else
                     {
-                        if (!type.IsFrameworkType && type.IsEnum)
+                        if (!type.IsFrameworkType && type.IsEnum && _inputModel.BaseModel.DiscriminatorProperty!.Type is InputEnumType inputEnumType)
```

Comment:
we do not have to add the `!` if we use `discriminator` here, could we do that?

Reply:
We need to get the InputType anyway when calling CreateEnum so we still need the `!`.